core/src/main/java/com/javadiscord/jdi/core/Discord.java

```diff
@@ -152,6 +156,27 @@ public void start() {
         EXECUTOR.execute(discordRequestDispatcher);
     }
 
+    public void stop() {
+        if (this.webSocketManager != null) {
+            this.webSocketManager.stop();
+        }
+
+        ThreadPoolExecutor threadPool = (ThreadPoolExecutor) EXECUTOR;
+        if (threadPool.getPoolSize() > 0) threadPool.shutdown();
+
+        try {
+            if (!threadPool.awaitTermination(30, TimeUnit.SECONDS)) {
+                threadPool.shutdownNow();
+                if (!threadPool.awaitTermination(30, TimeUnit.SECONDS)) {
+                    LOGGER.error("Threads not terminated properly");
+                }
+            }
+        } catch (InterruptedException ie) {
+            threadPool.shutdownNow();
+            Thread.currentThread().interrupt();
+        }
```

Comment:
I would prefer if you would introduce a new boolean in the `DiscordRequestDispatcher` which will be used in the while loop. This way you can properly shutdown it.

Reply:
So to understand your idea better, you mean to introduce that boolean which tracks in the `run()` method of `DiscordRequestDispatcher.java` if there has still been a request or not, and if not, to set the bool which then triggers the `stop()` method in `Discord.java` if e.g. bool is true due to no requests anymore?

Reply:
I wasn't really talking about the case of missing requests, though that might be a good idea to think about.
I was talking about the `while(true)` loop, it is better to use some `while(active)` loop and then setting the boolean to false for shutdown instead of causing an interrupt which in the current implementation even gets ignored.

Reply:
Pushed a new commit, hope i got your idea. [b0edfa9](https://github.com/javadiscord/java-discord-api/pull/100/commits/b0edfa9addc65932caeb8da5846fbcf556396cc7) 

Maybe it makes sense to add a "How to Run" this Discord Bot for other contributors. So that i can test the code better :-)!

Reply:
yeah we are kinda missing documentation